File: Python/Leetcode/1_two_sum.py
from typing import List

class Solution():
    def __init__(self):
        self.Dict = dict()

    def twoSum(self, nums: List[int], target: int) -> List[int]:
        for i,v in enumerate(nums):
            if v in self.Dict:
                return [self.Dict[v], i]
            else:
                self.Dict[target - v] = i
        return []

if __name__ == '__main__':
    print(f"{Solution().twoSum([2, 7, 11, 15], 9)}")
    print(f"{Solution().twoSum([3, 2, 4], 6)}")
    print(f"{Solution().twoSum([3, 3], 6)}")

File: Rust/leetcode/src/bin/1_two_sum.rs
use std::collections::HashMap;

pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
    let mut mapping: HashMap<i32, i32> = HashMap::new();

    for (i, x) in nums.iter().enumerate(){
        match mapping.get(x) {
            Some(&n) => return vec!(n, i as i32),
            None => mapping.insert(target - x, i as i32),
        };
    }

    vec![]
}

fn main() {
    println!("2 7 11 15 - 9 = {:?}", two_sum(vec!(2,7,11,15), 9));
    println!("2 11 7 15 - 9 = {:?}", two_sum(vec!(2,11,7,15), 9));
    println!("2 7 11 15 - 26 = {:?}", two_sum(vec!(2,7,11,15), 26));
}
